File: Java-Programs/Array/SecondHighest.java
//Write a Java Program to find the second-highest number in an array.

public class SecondHighest {
    public static void main(String[] args) {
        int arr[] = { 12, 43, 23, 11, 54, 16 };
        int SecondMax = 0;
        int Max = 0;
        int Len = arr.length;
        for (int i = 0; i < Len; i++) {
            if (arr[i] > Max) {
                SecondMax = Max;
                Max = arr[i];
            } else if (arr[i] > SecondMax) {
                SecondMax = arr[i];
            }
        }
        System.out.println("Second height number in array are: " + SecondMax);
        System.out.println("largest number in the array are: " + Max);
    }
}

File: PHP-Programs/Array/SecondHighest.php
<?php
//Write a Java Program to find the second-highest number in an array.
  $arr = [ 12, 43, 23, 11, 54, 16 ];
      $SecondMax = 0;
      $Max = 0;
      $Len = count($arr);
      for ($i = 0; $i < $Len; $i++) {
          if ($arr[$i] > $Max) {
              $SecondMax = $Max;
              $Max = $arr[$i];
          } elseif ($arr[$i] > $SecondMax) {
              $SecondMax = $arr[$i];
          }
      }
      echo "Second height number in array are:  $SecondMax\n";
      echo "largest number in the array are:  $Max\n";
?>



<!-- <?php
// another way
$array = [12, 43, 23, 11, 54, 16];
$secondMax = $max = PHP_INT_MIN;
$len = count($array);

foreach ($array as $element) {
    if ($element > $max) {
        $secondMax = $max;
        $max = $element;
    } elseif ($element > $secondMax) {
        $secondMax = $element;
    }
}

echo "Second highest number in the array is: $secondMax\n";
echo "Largest number in the array is: $max\n";
?> -->
